controller/k8s/api.go

```diff
@@ -246,7 +247,7 @@ func NewAPI(
 }
 
 // Sync waits for all informers to be synced.
-func (api *API) Sync(stopCh <-chan struct{}) {
+func (api *API) Sync(stopCh <-chan struct{}) error {
```

Comment:
this seems like a pretty invasive change, just so that we can print a customized error in the service mirror.  I think I'm -1 on this.